### PR TITLE
feature/create-admin-panel

### PR DESCRIPTION
**Pull request by rniedziolka**

### CODE REVIEW

File: app/javascript/packs/admin.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// This file is automatically compiled by Webpack, along with any other files
```
Comment (ciejj) [inline app/javascript/packs/admin.js:1]: Do we need this file?

File: app/assets/stylesheets/admin.css [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/*
+ * This is a manifest file that'll be compiled into application.css, which will include all the files
```
Comment (ciejj) [inline app/assets/stylesheets/admin.css:2]: Do we need this file?

File: app/views/admin/pages/dashboard.html.erb [hunk 3]
```diff
@@ -0,0 +1 @@
+Admin dashboard
```
Comment (ciejj) [inline app/views/admin/pages/dashboard.html.erb]: No newline at the end of file :)

File: app/views/layouts/admin.html.erb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          <h4>Navigation</h4>
+          <ul>
+             <% if !admin_user_signed_in? %>
+              <li><%= link_to 'Log in', new_admin_user_session_path %></li>
+             <% else %>
+              <li><%= link_to 'Log out', destroy_admin_user_session_path, method: :delete %></li>
+             <% end %>
+          </ul>
+          <hr />
+      </nav>
```
Comment (ciejj) [inline app/views/layouts/admin.html.erb]: I would suggest moving admin navigation bar to a partial

File: spec/system/kudo_spec.rb [hunk 5]
```diff
@@ -10,10 +10,12 @@
...
   it 'crud kudo' do
-    user = FactoryBot.create(:employee)
-    login_as(employee)
-
     visit root_path
+    click_link 'Login page'
+    fill_in 'Email', with: employee.email
+    fill_in 'Password', with: employee.password
+    click_button 'Log in'
+    expect(page).to have_content 'Signed in successfully.'
```
Comment (ciejj) [inline spec/system/kudo_spec.rb:18]: Consider using devise login helper here - you have already covered user logging with tests, so there is not necessary to check it here again.

https://webcache.googleusercontent.com/search?q=cache:prAqPB4X0REJ:https://sajadtorkamani.com/how-to-login-in-capybara-tests-when-using-devise/+&cd=3&hl=pl&ct=clnk&gl=pl

**Review by ciejj — APPROVED** [review]
Looks good! 😊 I have left some cosmetic comments.

File: app/javascript/packs/admin.js (hunk 1 above)
Comment (rniedziolka) [reply app/javascript/packs/admin.js:1]: Yes, I refer to this file in 18 line here:

app/views/layouts/admin.html.erb
18  <%= javascript_pack_tag 'admin' %>

without it, the code doesn't compile.

File: app/assets/stylesheets/admin.css (hunk 2 above)
Comment (rniedziolka) [reply app/assets/stylesheets/admin.css:2]: Yes, I refer to this file in 17 line here:

app/views/layouts/admin.html.erb

17    <%= stylesheet_link_tag 'admin', media: 'all' %>

without it, the code doesn't compile.
Perhaps I will change it at a later stage of the work, It wasn't a priority for me now.

File: app/views/admin/pages/dashboard.html.erb (hunk 3 above)
Comment (rniedziolka) [reply app/views/admin/pages/dashboard.html.erb]: Yesss... Will fix :P

File: app/views/layouts/admin.html.erb (hunk 4 above)
Comment (rniedziolka) [reply app/views/layouts/admin.html.erb]: Yes, I will refine it, I wanted to meet the assumptions of the task first, I will refine the aesthetics as soon as possible :)

File: spec/system/kudo_spec.rb (hunk 5 above)
Comment (rniedziolka) [reply spec/system/kudo_spec.rb:18]: I had a few rspec bugs in this round and applied some corrections, they are here for which I apologize.

File: app/assets/stylesheets/admin.css (hunk 2 above)
Comment (ciejj) [reply app/assets/stylesheets/admin.css:2]: Please don't get me wrong - I did not mean, that it is an issue.
It was only to make sure, that you are aware why is it here :)

**Comment by oskarlakner** [comment]
🙌

**Comment by rniedziolka** [comment]
Hi @oskarlakner and @ciejj,

I made a few adjustments, mainly to make the look pleasing to me, previously it was really so strict.

https://github.com/rniedziolka/employee-recognition-platform-1637255604/pull/11/commits/09c872c90ac46d5fe70d19a03978590c96153cea

I follow the next task so as not to waste time.

Have a great day guys!

File: app/views/pages/_headeradmin.html.erb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  <form class="container-fluid justify-content-start">
+   <div class="container">
+    
+    <% if admin_user_signed_in?  %>
+   
+    <%= link_to 'Home page', pages_home_path, class:"btn btn-outline-success me-2" %>
+    
+    <%= link_to 'Admin Panel', admin_root_path, class:"btn btn-outline-success me-2"%>
+    
+    <%= link_to 'Kudos', kudos_path, class:"btn btn-outline-success me-2" %>
```
Comment (rniedziolka) [inline app/views/pages/_headeradmin.html.erb]: I know that this line is beyond the scope of this task.

**Comment by rniedziolka** [comment]
@oskarlakner @ciejj any other suggestions?
Can I merge this one?

**Comment by ciejj** [comment]
Sure - its approved, and Oskar has blessed it as well :)

**Comment by rniedziolka** [comment]
Thank you guys again!

**Pull request merged**